[2492] Add new page to document installation methods (#3941)
psf/black: docs/usage_and_configuration/ways_to_install_black.md

# Ways to install Black

There are several ways you can install _Black_.

## Install from PyPI

> To install the latest release of _Black_ from [PyPI] (Requires Python 3.8+), use:

Reviewer: I wonder if we can link to the pyproject.toml directive here to avoid having _yet_ another place to upgrade when we move to requiring >= 3.9 - All for ideas here but can live with it if it's to hard ...

Author: I'm checking if this can be done via tomli and Sphinx configuration somehow.

> `pip install black`

Reviewer: O - Just thought of it - Please talk about all black's optional installs in this section too, please.

Author: Added, please check!

A _Black_ release currently offers three types of artifacts via PyPI, as outlined in the
[Release Process]:

1. The source distribution of the release
2. Generic Python wheel, meant for use on any Python supported platform
3. Platform and Python version specific wheels that offer significantly improved
   performance, compiled using [mypyc]

By default, `pip` will prefer a compatible wheel and revert to the source distribution

> if no such wheels are found, as outlined in [Python documentation].

Reviewer: ```suggestion
if no such wheels are found, as outlined in [Python documentation](LINK).
```
Did you forget the link?

Author: The links were all declared at the end of the document, similar to other pages in the documentation. I've changed all of them to avoid confusion.

## Install from GitHub

> To install the latest version of _Black_ from GitHub, use:
> 
> `pip install git+https://github.com/psf/black`

Reviewer: ```suggestion
To install the latest version of _Black_ from GitHub, use:

`pip install git+https://github.com/psf/black`

- This will install from the latest commit from `main` branch
```

## Get native binaries from GitHub

[GitHub Releases] for _Black_ contain self-contained, native binaries for multiple
platforms (built using PyInstaller). This allows you to download the executable for your
platform and run _Black_ without a Python runtime installed.

## Black Docker images

> Official _Black_ Docker images are available on [Docker Hub]. For more information,
> check the [Black Docker image] section.

Reviewer: ```suggestion
Official _Black_ Docker images are available on [Docker Hub](https://hub.docker.com/r/pyfound/black). For more information,
check the [Black Docker image](LINK_TO_PAGE) page.

- `docker run --rm --volume $(pwd):/src --workdir /src pyfound/black:latest_release black --check .`
```

Command example optional but I think we should fix the link to make this page more useful.

Author: I left out the command example because it's already present in the Black Docker page, can add it if required.
